posixlib/src/main/scala/scala/scalanative/posix/signal.scala

```diff
@@ -0,0 +1,411 @@
+package scala.scalanative
+package posix
+
+import scala.scalanative.unsafe._
+
+/**
+ * Some of the functionality described on this reference page extends the ISO C standard.
+ * Applications shall define the appropriate feature test macro (see XSH The Compilation Environment)
+ * to enable the visibility of these symbols in this header.
+ *
+ * Use clib macros and functions to get the full POSIX interface
+ *
+ * Note 1: The functionality described may be removed in a future version of this volume of POSIX.1-2017
+ * Note 2: The functionality is an option marked as XSI (X/Open System Interfaces)
+ *
+ * Note 1 or 2: followed by Linux or macOS means that the feature is not available on that platform with
+ * default compile options so the code is commented out. The code is commented out so the next developer
+ * that comes along will not have to rediscover these limitations.
+ */
+@extern
+object signal {
+  // define the following macros, which shall expand to constant expressions with distinct values
+  // that have a type compatible with the second argument to, and the return value of, the signal() function,
+  // and whose values shall compare unequal to the address of any declarable function.
+
+  // Note 1: Linux
+  // @name("scalanative_sig_hold")
+  // def SIG_HOLD: CFunctionPtr1[CInt, Unit] = extern
+
+  import sys.types
+  // pthread_t, size_t, and uid_t types as described in <sys/types.h>
+  type pthread_t = types.pthread_t
+  type size_t    = types.size_t
+  type uid_t     = types.uid_t
+
+  // timespec structure as described in <time.h>
+  type timespec = time.timespec
+
+  // define the following data types
+  type sig_atomic_t = CInt
+
+  // Integer or structure type of an object used to represent sets of signals
+  // macOS CUnsignedInt
+  // Linux CStruct1[CArray[CUnsignedLong, Nat.Digit[Nat._1, Nat._6]]]
+  type sigset_t = Ptr[Byte]
+
+  type pid_t          = types.pid_t
+  type pthread_attr_t = types.pthread_attr_t
+
+  type sigevent = CStruct5[
```

Comment:
This comment can be skipped if one does not want to go down
the rabbit hole.   Understood entirely that this is an area of complexity,
which is probably going to cause unending grief no matter how it is done.

We may both have other, urgent, areas of greater interest and with
a better cost/benefit ratio.

Since this is review mode, Here we go, after the White Rabbit...

Lines 65/66 define the type sigval.  That definition appears reasonable
but may change as it is exercised in the field.   For now lets go with it.

If those lines were moved to jujst before here,  (yes, I am old Pascal style, declare before use. Understood they they do not technically need to be moved), I believe the type `sigval` could be used in the `sigevent` notification 
rather than as `Ptr[sigval]`.  That is,  use the type rather than a Ptr
to the type.  Since we are doing `copy-in/copy-out` semantics anyway,
the underlying OS will not be writing directly into the SN structure.

IMO, removing the level of indirection (if possible) make it easier to 
trace the code.

Reply:
No, this has to be a `Ptr` and de-referenced in the C helper.